src/com/company/CricketMatch.java

```diff
@@ -0,0 +1,221 @@
+package com.company;
+import java.util.List;
+
+public class CricketMatch {
+    Player strikerBatsman;
+    Player nonStrikerBatsman;
+    Player strikeBowler;
+    Team battingTeam;
+    Team bowlingTeam;
+    Team firstBattingTeam;
+    Team secondBattingTeam;
+    private void swapPlayer()
+    {
+        Player temp = strikerBatsman;
+        strikerBatsman = nonStrikerBatsman;
+        nonStrikerBatsman = temp;
+    }
+
+    private void swapTeam()
+    {
+        Team temp = battingTeam;
+        battingTeam = bowlingTeam;
+        bowlingTeam = temp;
+    }
+
+    private void printWicketCommentary()
+    {
+        System.out.println("Wicket gone!");
+        System.out.println("--------------------------------------------------------------------------");
+        System.out.println(strikerBatsman.getName() + "\t" + strikerBatsman.getRole() + "\nRun - " + strikerBatsman.getRunScore() + "(" + strikerBatsman.getNoOfBallsPlayed() +")");
+        System.out.println("Team Score - " + battingTeam.getRun() + " \t Wickets - " + battingTeam.getWickets());
+        System.out.println("--------------------------------------------------------------------------");
+    }
+
+    private void printCommentary(int run)
+    {
+        System.out.println(run + "\t Total Run - " + battingTeam.getRun());
+    }
+
+    private void updateScoreBoard(int run)
+    {
+        strikerBatsman.addRunScore(run);
+        strikerBatsman.addNoOfBallsPlayed(1);
+        battingTeam.addRun(run);
+        strikeBowler.addGivenRun(run);
+        if(run % 2 == 1)swapPlayer();
+    }
+
+    private void wicketGone()
+    {
+        strikerBatsman.setBattingStatus(PlayerBattingStatus.status3.getValue());
+        strikerBatsman.addNoOfBallsPlayed(1);
+        battingTeam.addWickets(1);
+        strikeBowler.addNumberOfWicketsTaken(1);
+        printWicketCommentary();
+
+        if(battingTeam.getWickets() >= 10){
+            return;
+        }
+
+        if(strikerBatsman.getId() > nonStrikerBatsman.getId())
```

Comment:
order of batting should not be derived from id. if in future logic of id changes, then you will start fetching wrong player from team. Maintain a batting order in separate field.
We discussed this yesterday as well